include/oneapi/dpl/ranges

```diff
@@ -25,8 +27,8 @@
 #    define _ONEDPL_CXX_STANDARD_LIBRARY_CHECK 1
 #endif
 
-static_assert(__cplusplus >= 201703L && _ONEDPL_CXX_STANDARD_LIBRARY_CHECK,
-              "The use of the range-based API requires C++17 and the C++ standard libraries coming "
+static_assert(_ONEDPL_CXX_STANDARD_LIBRARY_CHECK,
```

Comment:
What is the reason we drop `__cplusplus >= 201703L` check? Is that because we believe it becomes useless after switch our minimal required version to C++17?